MacOS working docker-tizonia

dominic-codespoti opened the pull request:
Requires pulse audio to be installed via homebrew, and the following lines in /usr/local/Cellar/pulseaudio/13.0/etc/pulse/default.pa to be uncommented:
load-module module-esound-protocol-tcp
load-module module-native-protocol-tcp

Might've missed a step, but from my recollection my docker-tizonia file in-conjunction with these steps should allow MacOSX support, works on my device :)

dominic-codespoti committed:
MacOS working docker-tizonia

Requires pulse audio to be installed via homebrew, and the following lines in /usr/local/Cellar/pulseaudio/13.0/etc/pulse/default.pa to be uncommented:
load-module module-esound-protocol-tcp
load-module module-native-protocol-tcp

dominic-codespoti committed:
Update README.md

dominic-codespoti committed:
Update README.md

dominic-codespoti committed:
Update README.md

dominic-codespoti committed:
Update README.md

victor-ca commented on README.md:
```diff
@@ -4,6 +4,30 @@
 
 Containerized [**Tizonia**](http://www.tizonia.org/) cloud music player that uses the host's sound system.
 
+## Mac Support
```
Howdy
thanks for trying this out on mac; i can confirm that it works
i assembled the following script so we would automate these steps; hope it eases someone else's setup

```sh
#get pulse audio
brew install pulseaudio

#uncomment load-module module-esound-protocol-tcp and load-module  module-native-protocol-tcp
sed -i.bak -e 's/^#load-module module-esound-protocol-tcp/load-module module-esound-protocol-tcp/g' -e 's/^#load-module module-native-protocol-tcp/load-module module-native-protocol-tcp/g'  '/usr/local/Cellar/pulseaudio/13.0/etc/pulse/default.pa'

#list sinks with description
pactl list sinks | egrep '(^\s+(Name|Description).*$)|^Sink.*$'

echo 'Pick an output device enter the numer ( enter a number between 1 and' $(pactl list sinks short | wc -l) ')'
read outputChannel

pacmd set-default-sink $outputChannel
pulseaudio --load=module-native-protocol-tcp --exit-idle-time=-1 --daemon
cp ./docker-tizonia /usr/local/bin  

./docker-tizonia

chmod a+wrx $HOME/.config/tizonia
```

matthewarmand commented:
This PR probably is more likely to be merged if the `docker-tizonia` changes are done a bit differently so that the script can recognize whether its running under Mac or Linux and run the correct `docker run` command accordingly. 

For example, potential new `docker-tizonia`:
```bash
#!/bin/bash -e

USER_ID=$(id -u);
GROUP_ID=$(id -g);

if uname -s | grep -iq "Darwin" ; then
  pulse_server=docker.for.mac.localhost
  runtime_dir="$HOME"
else
  pulse_server=unix:"${XDG_RUNTIME_DIR}/pulse/native"
  runtime_dir="${XDG_RUNTIME_DIR}/pulse"
fi

docker run -it --rm \
    -e PULSE_SERVER="$pulse_server" \
    --volume="$runtime_dir":"$runtime_dir" \
    --volume="${HOME}/.config/tizonia":/home/tizonia/.config/tizonia \
    --volume="${HOME}/.config/pulse/cookie":/home/tizonia/.config/pulse/cookie \
    --name tizonia \
    tizonia/docker-tizonia "$@";

```

matthewarmand committed:
Merge branch 'master' of github.com:thed24/docker-tizonia-mac-support into mac-support-tweaks

matthewarmand committed:
Tweak the mac support a bit
- Re-organize the README a bit and pretty up the markdown a tad
- Make the docker-tizonia script smart enough to recognize whether its running under Mac or not and behave accordingly

matthewarmand committed:
Just a couple more README tweaks

dominic-codespoti committed:
Merge pull request #1 from matthewarmand/mac-support-tweaks

Mac support tweaks
